### ISSUE TITLE
Streamline our codebase

### ISSUE THREAD

**Issue by vitorbaptista** [okfn/datapackage-registry-js#9]
The compiled code is extremely big, given how little it does. As of https://github.com/okfn/datapackage-registry-js/commit/d4073c12fb9e11e47e44ed1d7c098e9b0fedb599, it has 570 KB (133 KB gzipped). Even after minifying with `uglifyjs --compress --mangle`, it still has 258 KB (71 KB gzipped). It includes the registry and its schemas, but it's still huge.

**Comment by vitorbaptista** [comment]
This will probably be useful https://github.com/samccone/The-cost-of-transpiling-es2015-in-2016

**Comment by pwalsh** [comment]
Is the problem we have one of transpilation? This code was not transpiled until this latest refactor. Is there a demonstrable bloat attributable to ES6?

**Comment by vitorbaptista** [comment]
Yes, but mostly because of the need for polyfills. The biggest culprits of the current size are:
- csv-parse: 85 KB minified (!!!);
- isomorphic-fetch: 7KB minified.

A smaller CSV parsing library would go a long way...

**Comment by pwalsh** [comment]
So we use this for client side CSV http://papaparse.com/ in most recent projects. that doesn't help with node.js though.

Also if this shows us that even on a very small lib like this, that ES6 creates too much bloat, we at least learn that at least in the case of isomorphic js apps, we should stick to ES5. LMK what you think.
